hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/FilterParams.java

```diff
@@ -17,45 +20,57 @@
  * Encapsulates a set of filters that may be specified (optionally). Properties
  * that are not specified (e.g. <code>null</code> for simple properties) When
  * applied, these filters are AND-gated.
- *
  */
 public class FilterParams {
 
     private final Collection<TargetUpdateStatus> filterByStatus;
     private final Boolean overdueState;
     private final String filterBySearchText;
-    private final Boolean selectTargetWithNoTag;
+    private final boolean selectTargetWithNoTag;
     private final String[] filterByTagNames;
     private final Long filterByDistributionId;
 
+    private FilterParams(@NotNull Long filterByDistributionId) {
```

Comment:
`@NotNull` is not working if you'll create the instance within this class [cf. line 69](https://github.com/eclipse/hawkbit/pull/827/files#diff-74763677ac56bf154da745b652646533R69)